repo-agent/repowatch/controllers/repowatch_controller.go

```diff
@@ -80,9 +136,48 @@ func NewGithubClient(ctx context.Context, k8sClient client.Client, repoWatch *re
 		githubConfig["email"] = string(secret.Data["email"])
 	}
 
-	ts := oauth2.StaticTokenSource(
-		&oauth2.Token{AccessToken: string(pat)},
-	)
+	clientID := os.Getenv("GITHUB_CLIENT_ID")
+	clientSecret := os.Getenv("GITHUB_CLIENT_SECRET")
+	refreshToken := string(secret.Data["refresh_token"])
+	expiryStr := string(secret.Data["expiry"])
+
+	var ts oauth2.TokenSource
+
+	if clientID != "" && clientSecret != "" && refreshToken != "" {
+		// Use refresh flow
+		oauthConf := &oauth2.Config{
```

Comment:
Is creating a new `oauth2.Config` on every reconciliation inefficient ? If this configuration is static for the lifetime of the controller, could we initialize this once when the controller starts and reuse the same config object in all reconciliations.

Does this create a separate http.Conn each time ?

Reply:
We need to refresh oauth token and config might be dynamic in future (change read-only -> read-write), leaving as is.